src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -499,6 +507,21 @@ private void walk(TypedVar symbol, boolean isDefault) {
       }
     }
 
+    private void walkDefaultInterface(FunctionType ftype) {
+      for (String propName : ftype.getOwnPropertyNames()) {
+        if (propName.equals("prototype")) continue;
+        JSType pType = ftype.getPropertyType(propName);
+        // Here we assume the enum is exported and handled separately.
+        if (pType.isEnumType()) continue;
+        emit("var");
+        emit(propName);
+        emit(":");
+        visitType(pType);
```

Comment:
Yes, the output goes from `var recordLike = {foo: A, bar: B}` to 
```
namespace NamespaceLike {
  var foo: A;
  var bar: B;
}
```

I am also removing visitRecordLike since it is not needed any longer. `walkDefaultInterface` can become `visitNamespaceLike` in case we find more usage for it.